tests/functional/event_handler/test_openapi_validation_middleware.py

```diff
@@ -330,3 +330,33 @@ def handler(user: Annotated[Model, Body(embed=True)]) -> Model:
     LOAD_GW_EVENT["body"] = json.dumps({"user": {"name": "John", "age": 30}})
     result = app(LOAD_GW_EVENT, {})
     assert result["statusCode"] == 200
+
+
+def test_validate_response_return():
+    # GIVEN an APIGatewayRestResolver with validation enabled
+    app = APIGatewayRestResolver(enable_validation=True)
+
+    class Model(BaseModel):
+        name: str
+        age: int
+
+    # WHEN a handler is defined with a body parameter
+    @app.post("/")
+    def handler(user: Annotated[Model, Body(embed=True)]) -> Response[Model]:
+        return Response(body=user, status_code=200)
+
+    LOAD_GW_EVENT["httpMethod"] = "POST"
+    LOAD_GW_EVENT["path"] = "/"
+    LOAD_GW_EVENT["body"] = json.dumps({"name": "John", "age": 30})
```

Comment:
for later.. but mutating this could cause problems when running tests in parallel (we don't **yet**).